addons/website/models/website_snippet_filter.py

```diff
@@ -86,14 +86,15 @@ def _prepare_values(self, limit=None, search_domain=None):
         if self.filter_id:
             filter_sudo = self.filter_id.sudo()
             domain = filter_sudo._get_eval_domain()
+            filter_context = literal_eval(filter_sudo.context)
             if 'website_id' in self.env[filter_sudo.model_id]:
                 domain = expression.AND([domain, self.env['website'].get_current_website().website_domain()])
             if 'is_published' in self.env[filter_sudo.model_id]:
                 domain = expression.AND([domain, [('is_published', '=', True)]])
             if search_domain:
                 domain = expression.AND([domain, search_domain])
             try:
-                records = self.env[filter_sudo.model_id].search(
+                records = self.env[filter_sudo.model_id].with_context(**filter_context).search(
```

Comment:
@rdeodoo Seems good to you? Not familiar with literal_eval

Reply:
This is correct, note that the variable is not very helpful here IMO
```diff
-records = self.env[filter_sudo.model_id].with_context(**filter_context).search(
+records = self.env[filter_sudo.model_id].with_context(**literal_eval(filter_sudo.context)).search(
```
Or move it next to the line where it is used in the `try`